taichi/transforms/auto_diff.cpp

```diff
@@ -1140,6 +1375,122 @@ class MakeAdjoint : public ADTransform {
     accumulate(stmt->val, insert<GlobalLoadStmt>(adjoint_ptr));
     stmt->parent->erase(stmt);
   }
+
+  void visit(MatrixPtrStmt *stmt) override {
+    if (stmt->origin->is<GlobalPtrStmt>() ||
+        stmt->origin->is<ExternalPtrStmt>()) {
+      // Handled in GlobalLoadStmt and GlobalStoreStmt
```

Comment:
Sorry I am not familiar with the usage of `MatrixPtrStmt`. Is it possible to use it to load a  `GlobalPtrStmt` or `ExternalPtrStmt`? And I am wondering why we skip them here?

Reply:
Thanks for the clarification.